solvers/fista_automatic_restart.py

```diff
@@ -0,0 +1,95 @@
+from benchopt import BaseSolver, safe_import_context
+
+with safe_import_context() as import_ctx:
+    import numpy as np
+    from scipy import sparse
+
+
+class Solver(BaseSolver):
+    name = 'FISTA_automatic_restart'
+    stopping_strategy = "callback"
+
+    references = [
+        'J.-F. Aujol, Ch. Dossal, H. Labarrière, A. Rondepierre, '
+        '"FISTA restart using an automatic estimation of the '
+        'growth parameter", HAL preprint : hal-03153525v4'
+    ]
+
+    def skip(self, X, y, lmbd, fit_intercept):
+        if fit_intercept:
+            return True, f"{self.name} does not handle fit_intercept"
+        return False, None
+
+    def set_objective(self, X, y, lmbd, fit_intercept):
+        self.X, self.y, self.lmbd = X, y, lmbd
+        self.fit_intercept = fit_intercept
+
+    def run(self, callback):
+        L = self.compute_lipschitz_constant()
+        n_features = self.X.shape[1]
+        w = np.zeros(n_features)
+        z = np.zeros(n_features)
+        C = 6.38  # Parameter controlling the restart frequency, 6.38 is the
+        # optimal theoretical value (we refer the user to the preprint for
+        # more details).
+        inner_iteration = 0
+        n_restarts = 0  # Restart counter
+        last_F = 0
+        ite_per_restart = np.array([int(2*C)])
+        F_tab = np.array([self.cost_function(w)])
```

Comment:
if `F_tab` is for tableau this is very French. Replace by `objectives` to be agnostic to math notation ?